lib/src/card_swiper.dart

```diff
@@ -267,31 +269,50 @@ class _CardSwiperState<T extends Widget> extends State<CardSwiper<T>>
     }
   }
 
-  //when the status of animation changes
-  void _animationStatusListener(AnimationStatus status) {
-    if (status == AnimationStatus.completed) {
-      setState(() {
-        if (_swipeType == SwipeType.swipe) {
-          widget.onSwipe?.call(_currentIndex, detectedDirection);
-          _stack.removeAt(_currentIndex);
+  // handle the onSwipe methode as well as removing the current card from the
+  // stack if onSwipe does not return false
+  void _handleOnSwipe() {
+    setState(() {
+      if (_swipeType == SwipeType.swipe) {
+        final cancelSwipe =
+            widget.onSwipe?.call(_currentIndex, detectedDirection) == false;
+
+        if (cancelSwipe) {
+          return;
+        }
```

Comment:
Can you make it so that the card animates back when we cancel the swipe?

Reply:
The `_animationStatusListener` function calls `_resetCardAnimation` at `AnimationStatus.completed`. So no other reset should be required as it is always reset at the end of the card animation, no matter what happens in between.